cpp/src/lists/contains.cu

```diff
@@ -44,384 +41,345 @@
- *                                        duplicate_find_option,
- *                                        rmm::cuda_stream_view,
- *                                        rmm::mr::device_memory_resource*)
- */
-std::unique_ptr<column> index_of(cudf::lists_column_view const& lists,
-                                 cudf::column_view const& search_keys,
+std::unique_ptr<column> index_of(lists_column_view const& lists,
+                                 column_view const& search_keys,
                                  duplicate_find_option find_option,
                                  rmm::cuda_stream_view stream,
                                  rmm::mr::device_memory_resource* mr)
 {
   CUDF_EXPECTS(search_keys.size() == lists.size(),
                "Number of search keys must match list column size.");
-
-  return search_keys.has_nulls()
-           ? cudf::type_dispatcher(search_keys.type(),
-                                   lookup_functor<true>{},  // Nulls in search keys
-                                   lists,
-                                   search_keys,
-                                   find_option,
-                                   stream,
-                                   mr)
-           : cudf::type_dispatcher(search_keys.type(),
-                                   lookup_functor<false>{},  // No nulls in search keys
-                                   lists,
-                                   search_keys,
-                                   find_option,
-                                   stream,
-                                   mr);
+  return cudf::type_dispatcher(search_keys.type(),
+                               dispatch_index_of{},
+                               lists,
+                               search_keys,
+                               search_keys.has_nulls(),
+                               find_option,
+                               stream,
+                               mr);
 }
 
-/**
- * @copydoc cudf::lists::detail::contains(cudf::lists_column_view const&,
- *                                        cudf::scalar const&,
- *                                        rmm::cuda_stream_view,
- *                                        rmm::mr::device_memory_resource*)
- */
-std::unique_ptr<column> contains(cudf::lists_column_view const& lists,
+std::unique_ptr<column> contains(lists_column_view const& lists,
                                  cudf::scalar const& search_key,
                                  rmm::cuda_stream_view stream,
                                  rmm::mr::device_memory_resource* mr)
 {
-  return to_contains(
-    index_of(lists, search_key, duplicate_find_option::FIND_FIRST, stream), stream, mr);
+  return to_contains(index_of(lists,
+                              search_key,
+                              duplicate_find_option::FIND_FIRST,
+                              stream,
+                              rmm::mr::get_current_device_resource()),
```

Comment:
Similar to the question below: Why be verbose about the choice of `mr`, when it's already in the defaults?